layout-server/src/main/java/org/finos/vuu/layoutserver/exceptions/GlobalExceptionHandler.java

```diff
@@ -1,40 +1,53 @@
 package org.finos.vuu.layoutserver.exceptions;
 
-import java.util.List;
-import java.util.NoSuchElementException;
-import java.util.stream.Collectors;
+import org.finos.vuu.layoutserver.dto.response.ErrorResponse;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.http.converter.HttpMessageNotReadableException;
 import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.annotation.ControllerAdvice;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
 
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
 @ControllerAdvice
 public class GlobalExceptionHandler {
 
     @ExceptionHandler(NoSuchElementException.class)
-    public ResponseEntity<String> handleNotFound(NoSuchElementException ex) {
-        return new ResponseEntity<>(ex.getMessage(),
-            org.springframework.http.HttpStatus.NOT_FOUND);
+    public ResponseEntity<Object> handleNotFound(HttpServletRequest request, Exception ex) {
+        return generateResponse(request, ex, HttpStatus.NOT_FOUND);
     }
 
     @ExceptionHandler({
-        HttpMessageNotReadableException.class,
-        MethodArgumentTypeMismatchException.class})
-    public ResponseEntity<String> handleBadRequest(Exception ex) {
-        return new ResponseEntity<>(ex.getMessage(),
-            org.springframework.http.HttpStatus.BAD_REQUEST);
+            HttpMessageNotReadableException.class,
+            MethodArgumentTypeMismatchException.class})
+    public ResponseEntity<Object> handleBadRequest(HttpServletRequest request, Exception ex) {
+
+        return generateResponse(request, ex, HttpStatus.BAD_REQUEST);
     }
 
     @ExceptionHandler(MethodArgumentNotValidException.class)
     public ResponseEntity<String> handleMethodArgumentNotValid(MethodArgumentNotValidException ex) {
         List<String> errors = ex.getFieldErrors()
-            .stream()
-            .map(fieldError -> fieldError.getField() + ": " + fieldError.getDefaultMessage())
-            .collect(Collectors.toList());
-
+                .stream()
+                .map(fieldError -> fieldError.getField() + ": " + fieldError.getDefaultMessage())
+                .collect(Collectors.toList());
         return new ResponseEntity<>(errors.toString(),
-            org.springframework.http.HttpStatus.BAD_REQUEST);
+                org.springframework.http.HttpStatus.BAD_REQUEST);
+    }
```

Comment:
This should be returning a generated error response

Reply:
I am happy to change this to use generateResponse() but note that if we do that we produce the following ErrorResponse: 
`{
    "timestamp": "2023-11-06T15:17:39.657+00:00",
    "status": 400,
    "error": "Bad Request",
    "message": "Validation failed for argument [0] in public org.finos.vuu.layoutserver.dto.response.LayoutResponseDto org.finos.vuu.layoutserver.controller.LayoutController.createLayout(org.finos.vuu.layoutserver.dto.request.LayoutRequestDto): [Field error in object 'layoutRequestDto' on field 'definition': rejected value [null]; codes [NotBlank.layoutRequestDto.definition,NotBlank.definition,NotBlank.java.lang.String,NotBlank]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [layoutRequestDto.definition,definition]; arguments []; default message [definition]]; default message [Definition must not be blank]] ",
    "path": "/api/layouts"
}`

where the errorMessage is 
`Validation failed for argument [0] in public org.finos.vuu.layoutserver.dto.response.LayoutResponseDto org.finos.vuu.layoutserver.controller.LayoutController.createLayout(org.finos.vuu.layoutserver.dto.request.LayoutRequestDto): [Field error in object 'layoutRequestDto' on field 'definition': rejected value [null]; codes [NotBlank.layoutRequestDto.definition,NotBlank.definition,NotBlank.java.lang.String,NotBlank]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [layoutRequestDto.definition,definition]; arguments []; default message [definition]]; default message [Definition must not be blank]] `
In the other PR we were building the errorMessage inside the handler and the message was looking like this:
`[definition: Definition must not be blank]`
which is a lot more concise and I liked better but obviously is not ideal because we are not wrapping it in a ErrorResponse like the other 400 responses. 
Maybe we can revisit the ErrorResponse constructor at some point to extract the error messages like we were doing in the handler and give us a more concise error message for this error.

Reply:
discussed offline: change generateResponse to extract error messages from excpetions

Reply:
done!

Reply:
Just to follow on from what we discussed earlier, I think one of the flaws of the overloading approach is that we could be creating some tight coupling between an 'error handler' and a 'generate response', so each error handler will end up needing it's own 'generate response' method. And in that case, we may as well not have the helper method, and handle each error message generation within the 'error handler'. Making 'generate response' the simple method (by just taking the error message rather than the exception) would resolve this, as we'd still only have one 'error handler', and a single 'generate response'.

Just something I thought of, but I'm happy to hear your thoughts.